app/src/main/java/com/github/libretube/util/LocaleHelper.kt

```diff
@@ -39,101 +41,50 @@ object LocaleHelper {
     private fun updateResourcesLegacy(context: Context, locale: Locale) {
         Locale.setDefault(locale)
         val resources: Resources = context.resources
-        val configuration: Configuration = resources.getConfiguration()
+        val configuration = resources.configuration
         configuration.locale = locale
-        resources.updateConfiguration(configuration, resources.getDisplayMetrics())
+        resources.updateConfiguration(configuration, resources.displayMetrics)
     }
 
-    fun getDetectedCountry(context: Context, defaultCountryIsoCode: String): String {
-        detectSIMCountry(context)?.let {
-            if (it != "") return it
-        }
-
-        detectNetworkCountry(context)?.let {
-            if (it != "") return it
-        }
-
-        detectLocaleCountry(context)?.let {
-            if (it != "") return it
+    private fun getDetectedCountry(context: Context): String {
+        return detectSIMCountry(context).ifEmpty {
+            detectNetworkCountry(context).ifEmpty {
+                detectLocaleCountry(context).ifEmpty { "UK" }
+            }
         }
-
-        return defaultCountryIsoCode
     }
 
-    private fun detectSIMCountry(context: Context): String? {
-        try {
-            val telephonyManager =
-                context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
-            return telephonyManager.simCountryIso
-        } catch (e: Exception) {
-            e.printStackTrace()
-        }
-        return null
+    private fun detectSIMCountry(context: Context): String {
```

Comment:
Instead of returning empty strings here, maybe we could let it return null instead and then do
```kotlin
private fun getDetectedCountry(context: Context): String {
     return detectSIMCountry(context) ?: detectNetworkCountry(context) ?: detectLocaleCountry(context) ?: "UK"
}
```

Reply:
The methods return empty strings if the country is not available, so those would have to be handled.